fedimint-client/src/sm/executor.rs

```diff
@@ -701,6 +807,7 @@ impl ExecutorBuilder {
 pub struct ActiveStateKey {
     // TODO: remove redundant operation id from state trait
     pub operation_id: OperationId,
+    // TODO: state being a key... seems ... risky?
```

Comment:
TODO: delete . I was surprised that we can't key `DynState` on something ... shorter. It forces me to pass `DynState` around a lot ... which isn't a big deal, I guess.

Reply:
@elsirion While intuitive I'd think that every module's sm would be unique ... if someone comes up with a use-case where it isn't ... it will be confusing for them what is happening. :D

Reply:
I agree that it's not optimal, longer term I'd like to key SMs the following way: `<operation_id> <sm_idx> <step_idx>`. I also want to get rid of the duplication between active and inactive state keys and instead keep a list of `<operation_id> <sm_idx>` prefixes for still-running SMs.

Reply:
Why it is a key: originally I assumed the DB had prefix notifications and I wanted to build the notification system around the serialized SMs in the DB.